### DISCUSSION TITLE
'NoneType' object has no attribute 'strip'   AND   Explicit valence for atom is greater than permitted

### DISCUSSION THREAD

**Discussion by RJ-Li** [RMeli/spyrmsd#132] [Q&A] [answered]
**Describe the bug**
```
[10:51:43] ERROR: Explicit valence for atom # 27 C, 6, is greater than permitted
Error calculating RMSD for 2019_general-train/13gs_1.sdf and 2019_general-sdf/13gs_ligand.sdf: 'NoneType' object has no attribute 'strip'
[10:51:43] Warning: molecule is tagged as 2D, but at least one Z coordinate is not zero. Marking the mol as 3D.
[10:51:43] Explicit valence for atom # 57 C, 6, is greater than permitted
```

**Environment**
 - OS: Linux
 - Python: py38
 - Package Manager [e.g. `pip`, `conda`]: conda

**Additional Context**
I do not know why these errors occurs and whether these errors are crucial or not

My code to calculate RMSD is:
```python
ref = io.loadmol(ref_lig)
dp = io.loadmol(dp_lig)

rmsd_value = rmsdwrapper(ref, dp)  # By default, strips hydrogens and use symmetry-correction
rmsd_float = float(rmsd_value[0])
```

**Comment by RMeli** [comment, marked as answer]
Hi @RJ-Li,

> [10:51:43] ERROR: Explicit valence for atom # 27 C, 6, is greater than permitted

This is an error from RDKit, which is unable to parse your input. It is telling you that the carbon atom at position 27 has a valence of 6, instead of the expected valence of 4. Therefore, I'd carefully check your input to see what is going on and where the issue is. The problem is not related with `spyrmsd` since it happens during parsing of the input file.

> Error calculating RMSD for 2019_general-train/13gs_1.sdf and 2019_general-sdf/13gs_ligand.sdf: 'NoneType' object has no attribute 'strip'

This a `spyrmsd` error but it is simply the consequence of the error above. If RDKit is unable to parse your input file and build a valid RDKit molecule, instead of returning a molecule it returns `None`. Therefore if `molecule = None`, calling `molecule.strip()` causes the error above. Fixing your input file will certainly get rid of this error too.

> [10:51:43] Warning: molecule is tagged as 2D, but at least one Z coordinate is not zero. Marking the mol as 3D.

This is a warning, also coming from RDKit. Again, it comes from a problem with your input file: the molecule is tagged as 2D, but not all Z coordinates are 0. Therefore, RDKit is telling you that it is treating it as a 3D molecule instead (despite the tag).

Whether you can ignore this warning depends on your upstream pipeline. There might or might not have been something wrong in your pre-processing, so it is worth checking visually that the input structure is indeed what you would expect.

> [10:51:43] Explicit valence for atom # 57 C, 6, is greater than permitted

This is the same issue above. Looks like you have a second carbon atom with a valence of 6 instead of the expected valence of 4.
